debugger/pythonExtension/gdb_sgx_plugin.py

```diff
@@ -10,24 +10,40 @@
 
 POINTER_SIZE = 8
 
-# These constant definitions must align with _oe_enclave structure defined in host\enclave.h
-OE_ENCLAVE_MAGIC_FIELD = 0
-OE_ENCLAVE_ADDR_FIELD = 2
-OE_ENCLAVE_HEADER_LENGTH = 0X28
-OE_ENCLAVE_HEADER_FORMAT = 'QQQQQ'
-OE_ENCLAVE_MAGIC_VALUE = 0x20dc98463a5ad8b8
-
-# The following are the offset of the 'debug' and
-# 'simulate' flag fields which must lie one after the other.
-OE_ENCLAVE_FLAGS_OFFSET = 0x788
-OE_ENCLAVE_FLAGS_LENGTH = 2
-OE_ENCLAVE_FLAGS_FORMAT = 'BB'
-OE_ENCLAVE_THREAD_BINDING_OFFSET = 0x28
-
-# These constant definitions must align with ThreadBinding structure defined in host\enclave.h
-THREAD_BINDING_SIZE = 0x38
-THREAD_BINDING_HEADER_LENGTH = 0X8
-THREAD_BINDING_HEADER_FORMAT = 'Q'
+# These constant definitions must align with oe_debug_enclave_t structure defined in debugrt/host.h
+class oe_debug_enclave_t:
+    O_MAGIC = 0
+    MAGIC_VALUE = 0xabc540ee14fa48ce
+    O_BASE_ADDRESS = 24
+    O_TCS = 32
+    O_NUM_TCS = 40
+    O_DEBUG = 48
+    O_SIMULATE = 49
+    O_NEXT = 56
+
+    def __init__(self, addr):
+        if addr:
+            self.magic = read_int_from_memory(addr + self.O_MAGIC, 8)
+        if not self.is_valid():
+            return
+        
+        self.base_address = read_int_from_memory(addr + self.O_BASE_ADDRESS, 8)
+        
+        self.tcs = []
```

Comment:
Consider making the field name plural for clarity.

Reply:
I have renamed it to tcs_array.